specification/paths/CarrierContracts-carrier_contract_id.json

```diff
@@ -39,5 +39,97 @@
         }
       }
     }
+  },
+  "patch": {
+    "tags": [
+      "CarrierContracts"
+    ],
+    "summary": "Update an carrier contract by id.",
+    "description": "This endpoint updates the carrier contract with the supplied id.",
+    "parameters": [
+      {
+        "$ref": "#/components/parameters/path-carrier_contract_id"
+      }
+    ],
+    "requestBody": {
+      "description": "The carrier contract object with attributes to be updated.",
+      "required": true,
+      "content": {
+        "application/vnd.api+json": {
+          "schema": {
+            "type": "object",
+            "required": [
+              "data"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "data": {
+                "allOf": [
+                  {
+                    "$ref": "#/components/schemas/CarrierContract"
+                  },
+                  {
+                    "required": [
+                      "id"
+                    ],
+                    "properties": {
+                      "attributes": {
+                        "properties": {
+                          "credentials": {
+                            "type": "string",
+                            "example": "{\"api_user\":\"your_api_user\",\"api_password\":\"your_password\"}"
+                          }
+                        }
+                      }
+                    }
+                  }
+                ]
+              }
+            }
+          }
+        }
+      }
+    },
+    "responses": {
+      "200": {
+        "description": "The carrier contract is updated.",
+        "content": {
+          "application/vnd.api+json": {
+            "schema": {
+              "type": "object",
+              "required": [
+                "data"
+              ],
+              "additionalProperties": false,
+              "properties": {
+                "data": {
+                  "$ref": "#/components/schemas/CarrierContract"
```

Comment:
For this, and other response bodies (in get all, get by id and post) the attributes: currency, name, status and active are specified as `not null` in the database. So you will always receive some value for these in a response. Therefore, they can be marked as required in the response body. You can achieve this with an "allOf".

Reply:
The status attribute is read only so that one we set internally in the Api